rp2040-hal/src/uart.rs

```diff
@@ -312,15 +312,15 @@ impl<D: UartDevice> UartPeripheral<Enabled, D> {
     /// - 0 bytes were read, a WouldBlock Error is returned
     /// - some bytes were read, it is deemed to be a success
     /// Upon success, the remaining slice is returned.
-    pub fn read_raw<'b>(&self, buffer: &'b mut [u8]) -> nb::Result<&'b mut [u8], ReadError<'b>> {
+    pub fn read_raw<'b>(&self, buffer: &'b mut [u8]) -> nb::Result<usize, ReadError<'b>> {
```

Comment:
I'd normally ask how you know that no-one is using this public function, but based on recent discussions it seems very unlikely that anyone is using this in any meaningful way